DatadogCore/Sources/Core/DatadogCore.swift

```diff
@@ -309,24 +308,21 @@ extension DatadogCore: DatadogCoreProtocol {
 internal struct DatadogCoreFeatureScope: FeatureScope {
     let contextProvider: DatadogContextProvider
     let storage: FeatureStorage
-    let telemetry: Telemetry
 
-    func eventWriteContext(bypassConsent: Bool, forceNewBatch: Bool, _ block: @escaping (DatadogContext, Writer) throws -> Void) {
-        // On user thread: request SDK context.
-        contextProvider.read { context in
-            // On context thread: request writer for current tracking consent.
-            let writer = storage.writer(
-                for: bypassConsent ? .granted : context.trackingConsent,
-                forceNewBatch: forceNewBatch
-            )
+    func eventWriteContext(bypassConsent: Bool, forceNewBatch: Bool, _ block: @escaping (DatadogContext, Writer) -> Void) {
+        // (on user thread) request SDK context
+        context { context in
+            // (on context thread) call the block
+            let writer = storage.writer(for: bypassConsent ? .granted : context.trackingConsent, forceNewBatch: forceNewBatch)
+            block(context, writer)
+        }
+    }
 
-            // Still on context thread: send `Writer` to EWC caller. The writer implements `AsyncWriter`, so
-            // the implementation of `writer.write(value:)` will run asynchronously without blocking the context thread.
-            do {
-                try block(context, writer)
-            } catch {
-                telemetry.error("Failed to execute feature scope", error: error)
-            }
```

Comment:
This change is intentional and makes the `block()` not throwing. Rationale is: the core being agnostic from any product-specific logic has no idea on how to handle or recover from the error. We had zero occurrence of `"Failed to execute feature scope"` in telemetry.